server/chat-commands/admin.ts

```diff
@@ -34,6 +34,21 @@ function bash(command: string, context: CommandContext, cwd?: string): Promise<[
 	});
 }
 
+function listMethods(obj: any) {
+	const methods = new Set<string>();
+	let current = obj;
+	do {
+		const curProps = Object.getOwnPropertyNames(current);
+		for (const prop of curProps) {
+			if (prop.includes('__') || prop.toLowerCase().includes('prop') || ['valueOf', 'constructor'].includes(prop)) {
+				continue;
+			}
+			methods.add(prop);
+		}
+	} while ((current = Object.getPrototypeOf(current)));
```

Comment:
Wait, if you want all properties, just use `keys` instead of `getOwnPropertyNames`...

Reply:
```ts
do {
	const curProps = Object.getOwnPropertyNames(current);
	for (const prop of curProps) {
		foo(prop);
	}
} while ((current = Object.getPrototypeOf(current)));
```

is a really long-winded way to do the exact same thing as

```ts
for (const prop in current) {
	foo(prop);
}
```

Reply:
Oh, I see, this is to include non-enumerable keys, because methods are non-enumerable by default now.